### DISCUSSION TITLE
The side effects in docs

### DISCUSSION THREAD

**Discussion by tyrickareo** [remix-run/remix#4211] [Q&A] [answered]
Remix version 1.7
    I am new to SSR also Remix, the [doc](https://remix.run/docs/en/v1/pages/gotchas#server-code-in-client-bundles) says that "For example, you can't import "fs-extra" directly into a route module", and the code:
```ts import { json } from "@remix-run/node"; 
import fs from "fs-extra";

export async function loader() {
  return json(await fs.pathExists("../some/path"));
}

export default function SomeRoute() {
  // ...
}
```

I do try, but It works well for me, there is not any errors and it returns `true` if path exists. And according to [this](https://remix.run/docs/en/v1/guides/constraints#no-module-side-effects). There is not any side effect in the example, cause `fs.pathExists` is wrapped in the loader. to my understand 
```ts
import { json } from "@remix-run/node"; 
import fs from "fs-extra";

//this should be side effect
fs.pathExists("../some/path");

export default function SomeRoute() {
  // ...
}
```
So what's my mistake, please let me know, thanks a lot

**Comment by sergiodxa** [comment, marked as answer]
The example lib should probably be changed, fs-extra has added sideEffects:false to their package.json (https://github.com/jprichardson/node-fs-extra/blob/0220eac966d7d6b9a595d69b1242ab8a397fba7f/package.json#L66) since then which mean now it's safer to import them in the route module itself instead of re-exporting from a `.server.ts` file.

Looks like they added it on this PR https://github.com/jprichardson/node-fs-extra/pull/941 made by the Remix team.

---

The important part is the `sideEffects:false` in the package.json, that tells the compiler the module doesn't have side effects, without that the compiler doesn't know and it will keep it on both client and server side bundles

**Reply by kiliman** [reply]
I agree. They should use the standard node `fs` package in that example. And then say instead of having to export and import from *fs.server.ts*, you can use `fs-extra` which has `sideEffects: false` and can be safely imported in your route module.
